### PR TITLE
feat: test service: replace plain Go map with sync.Map

### PR DESCRIPTION
**Pull request by panchoh**
To reduce contention on the map, we'll use:

https://golang.org/pkg/sync/#Map

### CODE REVIEW

File: tests/e2e/test_service.go [hunk 1]
```diff
@@ -102,23 +102,19 @@ func (a *alertStore) GetAll() []string {
...
 	}
 }
 
 func (s *snapStore) Get(version uint64) (v *protocol.SignedSnapshot, ok bool) {
-	s.Lock()
-	defer s.Unlock()
-	v, ok = s.d[version]
+	var tmpV interface{}
+	tmpV, ok = s.d.Load(version)
+	v = tmpV.(*protocol.SignedSnapshot)
```
Comment (aalda) [inline tests/e2e/test_service.go]: Can't you summarize it this way?

```
tmpV, ok = s.d.Load(version)
return tmpV.(*protocol.SignedSnapshot), ok
```

Comment (panchoh) [reply tests/e2e/test_service.go]: For that to work I would have to change the signature of the function, which currently reads:
`func (s *snapStore) Get(version uint64) (v *protocol.SignedSnapshot, ok bool) {`
That forces me to assign to `v` the return value.

I reckon that @gdiazlo wrote the signature like that to enhance readability.  Thoughts?

Comment (gdiazlo) [reply tests/e2e/test_service.go]: The compiler will optimize away the tmpV. This whole service will go to hell before ml ends, so I don't really care that much about its code style.

Comment (panchoh) [reply tests/e2e/test_service.go]: Ok, then.  I'll leave it as it is :-)

**Pull request merged**